recipes-extended/xen/files/libxl-xenmgr-support.patch

```diff
@@ -58,11 +58,16 @@ PATCHES
      xs_write(ctx->xsh, t, GCSPRINTF("%s/control/platform-feature-multiprocessor-suspend", dom_path), "1", 1);
      xs_write(ctx->xsh, t, GCSPRINTF("%s/control/platform-feature-xs_reset_watches", dom_path), "1", 1);
  
-@@ -1435,17 +1440,12 @@ static void domcreate_launch_dm(libxl__e
+@@ -1435,17 +1440,17 @@ static void domcreate_launch_dm(libxl__e
      {
          libxl__device_console console;
          libxl__device device;
 -        libxl_device_vkb vkb;
++
++        for (i = 0; i < d_config->num_vkbs; i++) {
```

Comment:
Instead of adding this here, this hunk can be dropped to re-instate the original code https://github.com/OpenXT/xenclient-oe/blob/4f925c48def5f07bb8f5e63ce5e07268e0d7a8fd/recipes-extended/xen/files/libxl-xenmgr-support.patch#L78
That is assuming 1 vkb is acceptable, but I think only one is added with vglass.

Reply:
This was a long time ago and I'm trying to remember why the vkbs ended up like this.  libxl-xenmgr-support.patch split apart the vfb and vkb creation in xl_parse.c because I think we wanted 2 vkbs (one for keyboard and one for mouse). I think you're right about vglass not needing 2 vkbs so we could drop the loop in xl_parse.c, but the UIVM still wants vfb=0, and vkb=1, so the initialization needs to remain split.

The ```d_config->vkbs[]``` array already has initialized vkbs from the code in xl_parse.c, so we're just calling the ```libxl__device_add()``` in this hunk instead of initializing another new vkb, like line 78 in libxl-xenmgr-support.patch.

Reply:
Yes, input server used two vkbs because a single one can only send either relative or absolute events.

UIVM is PVH, so it doesn't go through this code path and vkbs can be added separately anyway.  My suggested change would mean every HVM gets a vkb xenstore entry regardless of config.vkbd setting.  vglass actually providing a backend seems to be conditioned on the config.vglass-enabled setting.

The previously mentioned xenmgr change to use `vkb=[]` (https://github.com/jandryuk/manager/commits/vkb-fixup) may mean I'm missing something for the case without it.

Reply:
`vkb=[0|1]` and `vfb=[0|1]` are colliding with [xl.cfg(5)](https://xenbits.xen.org/docs/unstable/man/xl.cfg.5.html) definitions for `vkb=[ "VKB_SPEC_STRING", "VKB_SPEC_STRING", ...]` and `vfb=[ "VFB_SPEC_STRING", "VFB_SPEC_STRING", ...]` which, it sounds like, we could use directly without removing the other PV display support like @jandryuk points out.

Reply:
With @jandryuk changes the changes from `libxl-xenmgr-support.patch` related to `vkb` and `vfb` are dropped. Are there still concerns regarding this?